x-pack/platform/plugins/shared/automatic_import/public/components/create_integration/create_automatic_import/create_automatic_import.tsx

```diff
@@ -110,14 +110,16 @@ export const CreateAutomaticImport = React.memo(() => {
         <KibanaPageTemplate.Section grow paddingSize="l">
           {state.step === 1 && <ConnectorStep connector={state.connector} />}
           {state.step === 2 && <IntegrationStep integrationSettings={state.integrationSettings} />}
-          {state.step === 3 && (
-            <DataStreamStep
-              integrationSettings={state.integrationSettings}
-              celInputResult={state.celInputResult}
-              connector={state.connector}
-              isGenerating={state.isGenerating}
-            />
-          )}
+
+          {/* Persist step 3 and hide without unmounting as to preserve the EuiFilePicker State */}
```

Comment:
This seems to give a bug – the field here is not auto-filled any more (it was probably doing that on mount):

<img width="779" height="224" alt="image" src="https://github.com/user-attachments/assets/4afd6453-cfa2-4432-88f3-c3dbcd4d7f2c" />